spec/entities/base_spec.rb

```diff
@@ -51,37 +59,103 @@
-        response.expects(:body).returns({exists: false})
-        client.expects(:post).yields(request).returns(response)
-        KalibroClient::Entities::Base.expects(:client).with(any_parameters).returns(client)
-        expect(KalibroClient::Entities::Base.request('exists', {id: 1})[:exists]).to eq(false)
+    context 'when the record was not found' do
+      context 'and or the status is 404' do
+        let!(:faraday_stubs) {
+          Faraday::Adapter::Test::Stubs.new do |stub|
+            # stub.get receives arguments: path, headers, block
+            # The block should be a Array [status, headers, body]
+            stub.get('/bases/1') { |env| [404, {}, {}] }
+          end
+        }
+        let!(:connection) { Faraday.new {|builder| builder.adapter :test, faraday_stubs} }
+
+        before :each do
+          described_class.stubs(:client).returns(connection)
+        end
+
+        it 'is expected to raise a RecordNotFound error' do
+          expect { described_class.request(':id', {id: 1}, :get) }.to raise_error(KalibroClient::Errors::RecordNotFound)
+        end
+      end
+
+      context 'and or the response has a NotFound error message' do
+        let!(:faraday_stubs) {
+          Faraday::Adapter::Test::Stubs.new do |stub|
+            # stub.get receives arguments: path, headers, block
+            # The block should be a Array [status, headers, body]
+            stub.get('/bases/1') { |env| [422, {}, {'errors' => 'RecordNotFound'}] }
+          end
+        }
+        let!(:connection) { Faraday.new {|builder| builder.adapter :test, faraday_stubs} }
+
+        before :each do
+          described_class.stubs(:client).returns(connection)
+        end
+
+        it 'is expected to raise a RecordNotFound error' do
+          expect { described_class.request(':id', {id: 1}, :get) }.to raise_error(KalibroClient::Errors::RecordNotFound)
+        end
       end
     end
 
-    context 'with a children class from outside' do
-      class Child < KalibroClient::Entities::Base; end
+    # This uses a different method to stub faraday calls, that doesn't rely on stubbing particular methods of the requests.
+    # We should consider using it whenever possible instead of expectations.
+    context 'with an unsuccessful request' do
+      let!(:stubs) { Faraday::Adapter::Test::Stubs.new {|stub| stub.get('/bases/1/exists') { |env| [500, {}, {}] } } }
+      let(:connection) { Faraday.new {|builder| builder.adapter :test, stubs} }
 
-      it 'should successfully get the Kalibro version' do
-        response.expects(:body).returns({exists: false})
-        client.expects(:post).yields(request).returns(response)
-        Child.expects(:client).with(any_parameters).returns(client)
-        expect(Child.request('exists', {id: 1})[:exists]).to eq(false)
+      before :each do
+        subject.class.stubs(:client).returns(connection)
+      end
+
+      it 'should raise a RequestError with the response' do
```

Comment:
Using `should` for spec descriptions has been deprecated. Instead, `is expected to` fits better to the code.